standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -346,11 +346,12 @@ public void alterTable(RawStore msdb, Warehouse wh, String catName, String dbnam
             }
           }
           Deadline.checkTimeout();
+          Table table = msdb.getTable(catName, newDbName, newTblName);
           for (Entry<Partition, ColumnStatistics> partColStats : columnStatsNeedUpdated.entries()) {
             ColumnStatistics newPartColStats = partColStats.getValue();
             newPartColStats.getStatsDesc().setDbName(newDbName);
             newPartColStats.getStatsDesc().setTableName(newTblName);
-            msdb.updatePartitionColumnStatistics(newPartColStats, partColStats.getKey().getValues(),
+            msdb.updatePartitionColumnStatistics(table, newPartColStats, partColStats.getKey().getValues(),
```

Comment:
Hi @kgyrtkirk, should I create a specific ticket for this?

Reply:
or we can have it separately as well - right now I don't see the proposed changes of this PR because there were a lot of unreleated commits pushed yesterday